### PR TITLE
All components rendering

### PR DESCRIPTION
**Pull request by manuel12**
[BUG] All components are rendering to view #11

Ok so basically I removed the components that were located directly on .Section-app and added the routes there instead. this way now only the component that matches the route will be rendered in .Section-app, by default this will be the Dashboard component.

I also wrapped the header logo, settings and user menu icons with anchor tags that will redirect to "/", "/settings"  and "/profile" respectively. 

Finally I updated some css rules on Footer-footer a. These rules where directly on all <a> tags of the app, and would have a applied for example a margin-top of 18px to all links of the page. Now this will affect only the links inside the footer.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -36,7 +37,15 @@ const App = () => {
       <div className="container-fluid">
         <header className="App-header">
           <nav className="App-logo">
-            <img src="./logo512.png" width="192" height="192" alt="logo"></img>
+            <a href="/">
```
Comment (gbowne1) [inline src/App.js:40]: Logo must change state (size) from mobile to desktop? I think this is a thing that works, right?

File: src/App.js [hunk 2]
```diff
@@ -99,35 +110,35 @@ const App = () => {
...
+                <Routes>
+                  <Route exact path="/" element={<Dashboard />} />
+                  <Route exact path="/dashboard" element={<Dashboard />} />
+                  <Route exact path="/settings" element={<Settings />} />
+                  <Route exact path="/profile" element={<UserProfile />} />
+                  <Route exact path="/friends" element={<Friends />} />
+                  <Route exact path="/profile" element={<Profile />} />
+                  <Route exact path="/post" element={<Post />} />
+                  <Route exact path="/terms" element={<Terms />} />
+                </Routes>
```
Comment (gbowne1) [inline src/App.js:140]: There is no 404 Not Found path/route?  I believe its <Route path="*"... <NotFound>>

**Review by gbowne1 — CHANGES_REQUESTED** [review]
This Merge confilicts with App.js so unable to merge till that's fixed.

Needs a little more work but this is fine other than that.

Made some individual comments on the code as well.

**Review by gbowne1 — APPROVED** [review]
I wasn't originally sure we really need a footer.. but it's there.  Please make sure you add screen shots of working changes if at all possible.  This should fix the footer though.

File: src/App.js (hunk 2 above)
Comment (manuel12) [reply src/App.js:140]: You are right, let me re-add this.

File: src/App.js (hunk 1 above)
Comment (manuel12) [reply src/App.js:40]: Here I just added an <a> around the logo so it will redirect to "/" when clicked.

**Comment by manuel12** [comment]
Ok looks like the merge conflict is mostly due to the new TopNav component now holding the elements inside nav that were previously hardcoded on App.js. I will try solve the conflicts.

File: src/App.js (hunk 1 above)
Comment (gbowne1) [reply src/App.js]: yeah. It  should redirect to home (sometimes this is done with a HomeIcon somewhere in the topnav.  Somewhere I was reading about logo size state change when on mobile vs. desktop.. thats why react provided 192 and 512 sizes.  So the logo size should be dynamic.

File: src/App.js (hunk 2 above)
Comment (gbowne1) [reply src/App.js:63]: It should pick up the NotFound component which should render something in the view saying basically oopsie we couldnt find your page//route, etc. etc. etc.  I believe this got commented out.

**Review by gbowne1 — APPROVED** [review]

**Comment by gbowne1** [comment]
Merged this as confilcts were resolved but still 404 NotFound is missing.  Can re-PR with more changes. @manuel12

**Comment by manuel12** [comment]
With the current changes the header logo, settings and user menu icons will not have links to /, /settings and /profile currently, but I can open  an issue and add these to TopNav element.

Also I'm wondering about SideBar, I had a solution to add it before all the routes, like this:
```
          <SideNav />
          <main className="Main-app">
            <section className="Section-app">
              <Router>
                <Routes>
                  <Route exact path="/" element={<Dashboard />} />
                  <Route exact path="/dashboard" element={<Dashboard />} />
                  <Route exact path="/settings" element={<Settings />} />
                  <Route exact path="/profile" element={<UserProfile />} />
                  <Route exact path="/friends" element={<Friends />} />
                  <Route exact path="/profile" element={<Profile />} />
                  <Route exact path="/post" element={<Post />} />
                  <Route exact path="/terms" element={<Terms />} />
                </Routes>
              </Router>
            </section>
```
My reason to take the SideNav component outside of the routes is because as far as I understand SideNav should appear, on all pages regardless of route. So by taking the SideNav component out of the router part we achieve this effect of having it on all pages.

SideNav is currently written into the routes like this:
```
<Route path="*" element={<SideNav isSideNavVisible={isSideNavVisible} />} />
```

Instead I think we should use the <Route path="*" for the NotFound component. I can open an issue and update the SideNav and add the NotFound component if needed.

**Comment by gbowne1** [comment]
"as far as I understand SideNav should appear, on all pages regardless of route. "

Yes.  It gets put away with the click of the X.. but should be visible regaurdless of route.

Not found is normally written something like <Route path "*" element=<NotFound /> or whatever the component got named..  

Some things had to be done in our early work to get stuff rendered and designed.  Now that we have contributors this is less important.  A lot of things I was not initially making 

"... but I can open an issue and add these to TopNav element."  Ok go for it. :+1:  :100:

**Pull request merged**